### PR TITLE
Keras to tflite version of tests

### PR DESCRIPTION
**Pull request by rnugmanx**
### Tickets:
 - CVS-99807

### CODE REVIEW

File: tests/layer_tests/common/tf2_layer_test_class.py [hunk 1]
```diff
@@ -71,3 +99,30 @@ def get_framework_results(self, inputs_dict, model_path):
...
+        interpreter.allocate_tensors()
+        input_details = interpreter.get_input_details()
+        output_details = interpreter.get_output_details()
+
+        input_name_to_id_mapping = {input['name']: input['index'] for input in input_details}
+
+        for layer, data in inputs_dict.items():
+            tensor_index = input_name_to_id_mapping[layer]
+            tensor_id = next(i for i, tensor in enumerate(input_details) if tensor['index'] == tensor_index)
+            interpreter.set_tensor(input_details[tensor_id]['index'], data)
```
Comment (jane-intel) [inline tests/layer_tests/common/tf2_layer_test_class.py:115]:
```suggestion
            interpreter.set_tensor(input_details[tensor_id]['index'], data.astype(tensor['dtype']))
```

File: tests/layer_tests/common/utils/tf_utils.py [hunk 2]
```diff
@@ -135,3 +163,49 @@ def permute_nchw_to_nhwc(shape, use_new_frontend=False):
...
+def transpose_nhwc_to_nchw(data, use_new_frontend, use_old_api):
+    if use_new_frontend or not use_old_api:
+        return data
+
+    if len(data.shape) == 4:  # reshaping for 4D tensors
+        return data.transpose(0, 3, 1, 2)  # 2, 0, 1
+    elif len(data.shape) == 5:  # reshaping for 5D tensors
+        return data.transpose(0, 4, 1, 2, 3)  # 3, 0, 1, 2
+    else:
+        return data
```
Comment (rkazants) [inline tests/layer_tests/common/utils/tf_utils.py:189]: This routines not necessarily use for TFLite only. It makes sense to put them to the common utils then.

File: tests/layer_tests/common/utils/tf_utils.py [hunk 3]
```diff
@@ -135,3 +163,49 @@ def permute_nchw_to_nhwc(shape, use_new_frontend=False):
...
+
+
+def save_to_pb(tf_model, path_to_saved_tf_model):
+    tf.io.write_graph(tf_model, path_to_saved_tf_model, 'model.pb', False)
+    assert os.path.isfile(os.path.join(path_to_saved_tf_model, 'model.pb')), "model.pb haven't been saved " \
+                                                                             "here: {}".format(path_to_saved_tf_model)
+    return os.path.join(path_to_saved_tf_model, 'model.pb')
```
Comment (rkazants) [inline tests/layer_tests/common/utils/tf_utils.py:196]: absolutely this is not TFLite specific.

**Review by rkazants — CHANGES_REQUESTED** [review]

File: tests/layer_tests/common/utils/tf_utils.py [hunk 4]
```diff
@@ -76,6 +77,33 @@ def collect_tf_references(model_path, feed_dict, out_layer, output_nodes_for_fre
...
+    for output in output_details:
+        tf_lite_result[output['name']] = interpreter.get_tensor(output['index'])
+
+    result = dict()
+    for out in tf_lite_result.keys():
+        _tf_res = tf_lite_result[out]
+        result[out] = transpose_nhwc_to_nchw(_tf_res, use_new_frontend,
+                                             use_old_api)
+
+    return tf_lite_result
```
Comment (rkazants) [inline tests/layer_tests/common/utils/tf_utils.py:104]: propose to put all TF Lite specific routines into `tf_lite_layer_test_class.py`

File: tests/layer_tests/common/tf2_layer_test_class.py [hunk 5]
```diff
@@ -15,13 +17,38 @@ def save_to_tf2_savedmodel(tf2_model, path_to_saved_tf2_model):
...
+    converter = tf.compat.v1.lite.TFLiteConverter.from_saved_model(savedmodel)
+    tflite_model = converter.convert()
+
+    tflite_model_path = os.path.join(os.path.dirname(savedmodel), 'model.tflite')
+    with tf.io.gfile.GFile(tflite_model_path, 'wb') as f:
+        f.write(tflite_model)
+
+    return tflite_model_path
+
+
```
Comment (rkazants) [inline tests/layer_tests/common/tf2_layer_test_class.py:32]: please move this into TF Lite specific file.

**Review by rkazants — DISMISSED** [review]
PR breaks layer tests for TF FE.
Please fix it.

**Review by jane-intel — APPROVED** [review]

**Review by rkazants — APPROVED** [review]

**Pull request merged**